app/components/ConnectButton.tsx

```diff
@@ -6,15 +6,15 @@ import { useBalance, useEnsAvatar } from 'wagmi';
 
 import { ChainId } from '@/constants';
 import { Address, formatEther } from 'viem';
-import { Button, ButtonProps, ButtonSizeProp } from '@swapr/ui';
+import { Button, ButtonProps } from '@swapr/ui';
 import { PropsWithChildren } from 'react';
 import { cx } from 'class-variance-authority';
 
 interface CustomConnectButtonProps {
   address: Address;
   onClick: () => void;
   ensName?: string;
-  size?: ButtonSizeProp;
+  size?: ButtonProps['size'];
```

Comment:
why like this, seems a bit odd or is it the common pattern?

Reply:
It's part of the work done in https://github.com/SwaprHQ/swapr-ui/pull/24, we had some discussions offline with @berteotti on how to handle the task, and we decided that the cleanest and simplest way to do it was to have a single source of truth being the ButtonProps interface and then consume the properties on demand.